src/mesh/gmsh_reader.cpp

```diff
@@ -543,92 +713,250 @@ gmsh_hierarchic_to_lexicographic(const unsigned int degree)
-          for (unsigned int j = 0; j < dofs_per_line; ++j)
-            h2l[next_index++] = (j + 1) * n * n + i + 1;
-        // face 3, note the orientation!
-        for (unsigned int i = 0; i < dofs_per_line; ++i)
-          for (unsigned int j = 0; j < dofs_per_line; ++j)
-            h2l[next_index++] = (j + 1) * n * n + n * (n - 1) + i + 1;
-        // face 4
-        for (unsigned int i = 0; i < dofs_per_line; ++i)
-          for (unsigned int j = 0; j < dofs_per_line; ++j)
-            h2l[next_index++] = n * (i + 1) + j + 1;
-        // face 5
-        for (unsigned int i = 0; i < dofs_per_line; ++i)
-          for (unsigned int j = 0; j < dofs_per_line; ++j)
-            h2l[next_index++] = (n - 1) * n * n + n * (i + 1) + j + 1;
-
-        // inside hex
-        for (unsigned int i = 0; i < dofs_per_line; ++i)
-          for (unsigned int j = 0; j < dofs_per_line; ++j)
-            for (unsigned int k = 0; k < dofs_per_line; ++k)
-              h2l[next_index++] = n * n * (i + 1) + n * (j + 1) + k + 1;
+        h2l[next_index++] = 0;                          // 0
+        h2l[next_index++] = (1) * degree;               // 1
+        h2l[next_index++] = (n + 1)*degree;             // 2
+        h2l[next_index++] = (n) * degree;               // 3
+        h2l[next_index++] = (n * n) * degree;           // 4
+        h2l[next_index++] = (n * n + 1) * degree;       // 5
+        h2l[next_index++] = (n * n + n + 1) * degree;   // 6
+        h2l[next_index++] = (n * n + n) * degree;       // 7
+
+        if (degree > 1) {
+
+            face_nodes = face_node_finder(degree-2,mesh_reader_verbose_output);                     //PHYSICAL INTERPRETATION OF THE NODES
+
+//            if(mesh_reader_verbose_output) pcout << "DEGREE " << degree << std::endl;
+//            {
+//                unsigned int n = degree - 1;
+//                if(mesh_reader_verbose_output) pcout << "GMSH H2L " << std::endl;
+//                for (int j = n - 1; j >= 0; --j) {
+//                    for (unsigned int i = 0; i < n; ++i) {
+//                        const unsigned int ij = ij_to_num(i, j, n);
+//                        if(mesh_reader_verbose_output) pcout << face_nodes[ij] << " ";
+//                    }
+//                    if(mesh_reader_verbose_output) pcout << std::endl;
+//                }
+//            }
+//
+//            if(mesh_reader_verbose_output) pcout << "" << std::endl;
+
+
+            // line 0
+            for (unsigned int i = 0; i < dofs_per_line; ++i) {
+                h2l[next_index++] = i + 1;
+//                if(mesh_reader_verbose_output) pcout << "line 0 - " << i + 1 << std::endl;
+            }
 
-        Assert(next_index == dofs_per_cell, dealii::ExcInternalError());
+            // line 1
+            for (unsigned int i = 0; i < dofs_per_line; ++i) {
+                h2l[next_index++] = (i + 1) * n;
+//                if(mesh_reader_verbose_output) pcout << "line 1 - " << (i + 1) * n << std::endl;
```

Comment:
I notice that you only use this parameter in the periodic cube case and not in the NACA0012 or Gaussian bump cases. I think it would be nice for all gmsh reader functions to use this parameter for debugging purposes. Up to you, though, you're more familiar with what might cause issues with these new functions!

Reply:
So I think its easier if we leave it up to who ever is running those cases to use the verbose flag input for their cases if they want down the road, resolving this comment